File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/dataset/search/ContentCopySearchCriteria.java
/*
 * Copyright 2016 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.dataset.search;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.AbstractObjectSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchCriteriaToStringBuilder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.externaldms.id.IExternalDmsId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author anttil
 */
@JsonObject("as.dto.dataset.search.ContentCopySearchCriteria")
public class ContentCopySearchCriteria extends AbstractObjectSearchCriteria<IExternalDmsId>
{

    private static final long serialVersionUID = 1L;

    public ExternalDmsSearchCriteria withExternalDms()
    {
        return with(new ExternalDmsSearchCriteria());
    }

    public ExternalCodeSearchCriteria withExternalCode()
    {
        return with(new ExternalCodeSearchCriteria());
    }

    public PathSearchCriteria withPath()
    {
        return with(new PathSearchCriteria());
    }

    public GitCommitHashSearchCriteria withGitCommitHash()
    {
        return with(new GitCommitHashSearchCriteria());
    }

    public GitRepositoryIdSearchCriteria withGitRepositoryId()
    {
        return with(new GitRepositoryIdSearchCriteria());
    }

    @Override
    protected SearchCriteriaToStringBuilder createBuilder()
    {
        SearchCriteriaToStringBuilder builder = super.createBuilder();
        builder.setName("CONTENT_COPY");
        return builder;
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/dataset/search/ContentCopySearchCriteria.js
/**
 * @author anttil
 */
define([ "require", "stjs", "as/dto/common/search/AbstractObjectSearchCriteria", "as/dto/dataset/search/ExternalDmsSearchCriteria", "as/dto/dataset/search/ExternalCodeSearchCriteria", "as/dto/dataset/search/PathSearchCriteria", "as/dto/dataset/search/GitCommitHashSearchCriteria",
	"as/dto/dataset/search/GitRepositoryIdSearchCriteria", "as/dto/common/search/AbstractCompositeSearchCriteria" ], function(require, stjs, AbstractObjectSearchCriteria) {
	var ContentCopySearchCriteria = function() {
		AbstractObjectSearchCriteria.call(this);
	};
	stjs.extend(ContentCopySearchCriteria, AbstractObjectSearchCriteria, [ AbstractObjectSearchCriteria ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.dataset.search.ContentCopySearchCriteria';
		constructor.serialVersionUID = 1;

		prototype.withExternalDms = function() {
			var ExternalDmsSearchCriteria = require("as/dto/dataset/search/ExternalDmsSearchCriteria");
			return this.addCriteria(new ExternalDmsSearchCriteria());
		};

		prototype.withExternalCode = function() {
			var ExternalCodeSearchCriteria = require("as/dto/dataset/search/ExternalCodeSearchCriteria");
			return this.addCriteria(new ExternalCodeSearchCriteria());
		};

		prototype.withPath = function() {
			var PathSearchCriteria = require("as/dto/dataset/search/PathSearchCriteria");
			return this.addCriteria(new PathSearchCriteria());
		};

		prototype.withGitCommitHash = function() {
			var GitCommitHashSearchCriteria = require("as/dto/dataset/search/GitCommitHashSearchCriteria");
			return this.addCriteria(new GitCommitHashSearchCriteria());
		};

		prototype.withGitRepositoryId = function() {
			var GitRepositoryIdSearchCriteria = require("as/dto/dataset/search/GitRepositoryIdSearchCriteria");
			return this.addCriteria(new GitRepositoryIdSearchCriteria());
		};

	}, {
		criteria : {
			name : "Collection",
			arguments : [ "ISearchCriteria" ]
		}
	});
	return ContentCopySearchCriteria;
})